inst/templates/ad_advs.R

```diff
@@ -0,0 +1,432 @@
+  bind_rows(cdc_htage %>%
+    mutate(
+      SEX = case_when(
+        SEX == 1 ~ "M",
+        SEX == 2 ~ "F",
+        TRUE ~ NA_character_
+      ),
+      AGE = AGE * 30.4375
+    )) %>%
+  # AGEU is added in metadata, required for derive_params_growth_age
+  mutate(AGEU = "DAYS") %>%
+  arrange(AGE, SEX)
+
+## WEIGHT for age ----
+# Default reference sources: WHO for children <2 yrs old (< 730.5 days),
+# and CDC for children >=2 yrs old (>= 730.5 days)
+data(who_wt_for_age_boys)
+data(who_wt_for_age_girls)
+data(cdc_wtage)
+
+weight_for_age <- who_wt_for_age_boys %>%
+  filter(Day < 730.5) %>%
+  mutate(SEX = "M") %>%
+  bind_rows(who_wt_for_age_girls %>%
+    filter(Day < 730.5) %>%
+    mutate(SEX = "F")) %>%
+  rename(AGE = Day) %>%
+  bind_rows(cdc_wtage %>%
+    mutate(
+      SEX = case_when(
+        SEX == 1 ~ "M",
+        SEX == 2 ~ "F",
+        TRUE ~ NA_character_
+      ),
+      AGE = AGE * 30.4375
+    )) %>%
+  # AGEU is added in metadata, required for derive_params_growth_age
+  mutate(AGEU = "DAYS") %>%
+  arrange(AGE, SEX)
+
+## WHO - HEAD CIRCUMFERENCE for age ----
+data(who_hc_for_age_boys)
+data(who_hc_for_age_girls)
+
+who_hc_for_age <- who_hc_for_age_boys %>%
+  mutate(SEX = "M") %>%
+  bind_rows(who_hc_for_age_girls %>%
+    mutate(SEX = "F")) %>%
+  rename(AGE = Day) %>%
+  # AGEU is added in metadata, required for derive_params_growth_age
+  mutate(AGEU = "DAYS") %>%
+  arrange(AGE, SEX)
+
+## WHO - WEIGHT for HEIGHT/HEIGHT ----
+data(who_wt_for_ht_boys)
+data(who_wt_for_ht_girls)
+data(who_wt_for_lgth_boys)
+data(who_wt_for_lgth_girls)
+
+who_wt_for_ht_lgth <- who_wt_for_ht_boys %>%
+  mutate(SEX = "M") %>%
+  bind_rows(who_wt_for_ht_girls %>%
+    mutate(SEX = "F")) %>%
+  mutate(MEASURE = "HEIGHT") %>%
+  rename(HEIGHT_LENGTH = Height) %>%
+  bind_rows(who_wt_for_lgth_boys %>%
+    mutate(SEX = "M") %>%
+    bind_rows(who_wt_for_lgth_girls %>%
+      mutate(SEX = "F")) %>%
+    mutate(MEASURE = "LENGTH") %>%
+    rename(HEIGHT_LENGTH = Length))
+
+# Load source datasets ----
+
+# Use e.g. `haven::read_sas()` to read in .sas7bdat, or other suitable functions
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+
+# Once available to the Pharmaverse, needs to uncomment below data calls
+# data("vs_peds")
+# data("adsl_peds")
+
+vs <- vs_peds
+adsl <- adsl_peds %>% select(-DOMAIN)
+
+# When SAS datasets are imported into R using haven::read_sas(), missing
+# character values from SAS appear as "" characters in R, instead of appearing
+# as NA values. Further details can be obtained via the following link:
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values # nolint
+
+vs <- convert_blanks_to_na(vs)
+
+# Lookup tables ----
+# Assign PARAMCD, PARAM, and PARAMN
+param_lookup <- tibble::tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 1,
+  "HEIGHT", "HEIGHT", "Height (cm)", 2,
```

Comment:
Should LENGTH also be included in the lookup table?

Reply:
@Minlei0201 in our template (as per the outline doc we shared on our admiralpeds MS teams) we only use the HEIGHT parameter here as default but in the vignette we'll explain around body length. Most CRFs only likely collect as height in one parameter and then it depends on the child's age what they actually collect here.

Reply:
I see, thanks @rossfarrugia ! So if a study collects length, it will be bind with the height, and mapped to `PARAMCD="HEIGHT`"?

Reply:
for the sake of our test data and examples yes, but whatever the study mapped this as they can feed it into the functions using the `parameter` argument so the user has complete flexibility here.